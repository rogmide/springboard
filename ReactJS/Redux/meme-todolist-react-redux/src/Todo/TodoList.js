import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import "./TodoList.css";
import { v4 as uuidv4 } from "uuid";
import { useSelector } from "react-redux";

const TodoList = () => {
  const todos = useSelector((store) => store.todos);

  return (
    <>
      {" "}
      <div className="TodoList">
        <h1>Todo List</h1>
        {todos.map((t) => (
          <div key={t.id}>
            {/* {isEditForm === t.id && (
              <EditTodoForm todo={t} editTodo={editTodo} />
            )} */}
            <Todo id={t.id} text={t.text} done={t.done} scrach={t.scrach} />
          </div>
        ))}
      </div>
      <NewTodoForm />
    </>
  );
};

export default TodoList;
