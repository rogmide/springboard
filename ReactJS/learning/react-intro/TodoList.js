const TodoList = (props) => {
  return (
    <div>
      <h4>Todo List</h4>
      <ul>
        {props.todos.map((t, i) => (
          // NEVER USER THIS TO SETUP THE KEY, INDEX IS NOT GOOD AS KEY
          <li key={i}>
            <input type="checkbox" />
            <b>{t}</b>
          </li>
        ))}
      </ul>
    </div>
  );
};
