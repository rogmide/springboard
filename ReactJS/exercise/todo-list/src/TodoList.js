import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import "./TodoList.css";
import { v4 as uuidv4 } from "uuid";

const TodoList = () => {
  const INITIAL_STATE = [
    // default box to work with app
    { id: uuidv4(), text: "Walk the Dog", done: false, scrach: false },
  ];

  const [todos, setTodos] = useState(INITIAL_STATE);

  // Add box to the Array
  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, { ...newTodo, id: uuidv4() }]);
  };

  // Remove box from the Array
  const removeTodo = (id) => {
    setTodos(todos.filter((t) => t.id !== id));
  };

  // Scrach Todo
  const scrachTodo = (id) => {
    setTodos(
      todos.filter((t) => {
        if (t.id !== id) {
          t.scrach = true;
        }
      })
    );
  };
  return (
    <>
      {" "}
      <div className="TodoList">
        <h1>Todo List</h1>
        {todos.map(({ id, text, done, scrach }) => (
          <Todo
            removeTodo={removeTodo}
            id={id}
            text={text}
            done={done}
            scrach={scrach}
            key={id}
          />
        ))}
      </div>
      <NewTodoForm addTodo={addTodo} />
    </>
  );
};

export default TodoList;
