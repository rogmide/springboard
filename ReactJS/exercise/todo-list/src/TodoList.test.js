import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("render without crashing", function () {
  render(<TodoList />);
});

it("should match snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should remove/add a new todo", () => {
  const { queryByText, getByLabelText, queryByTestId } = render(<TodoList />);

  // Selecting
  const input = getByLabelText("Todo:");
  const btnAddTodo = queryByText("Add Todo");

  // Make sure that a Box is on Screen
  const btnTrashCan = queryByTestId("todo_trash");
  expect(btnTrashCan).toBeInTheDocument();

  // Click and check that is not there anymore
  fireEvent.click(btnTrashCan);
  expect(btnTrashCan).not.toBeInTheDocument();

  // Setup Next Box Form to be added
  fireEvent.change(input, { target: { value: "New Todo Is Added" } });

  // Click to Add New Todo
  fireEvent.click(btnAddTodo);

  // Make sure that a new Todo is on Screen
  expect(queryByTestId("todo_trash")).toBeInTheDocument();
});
