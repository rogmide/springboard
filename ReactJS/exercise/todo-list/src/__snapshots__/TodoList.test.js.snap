// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="TodoList"
  >
    <h1>
      Todo List
    </h1>
    <div
      class="Todo"
    >
      <p>
        Walk the Dog - 
        <span
          data-testid="todo_trash"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash-can "
            data-icon="trash-can"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
              fill="currentColor"
            />
          </svg>
        </span>
         
      </p>
    </div>
  </div>
  <h1>
    Add New Todo
  </h1>
  <form
    class="NewTodoForm"
  >
    <label
      for="text1"
    >
      Todo: 
    </label>
    <input
      id="text1"
      name="text"
      placeholder="Todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
